packages/web/src/react.tsx

```diff
@@ -25,14 +25,27 @@ import type { AnalyticsProps } from './types';
  * }
  * ```
  */
-function Analytics({
-  beforeSend,
-  debug = true,
-  mode = 'auto',
-}: AnalyticsProps): null {
+function Analytics(
+  props: AnalyticsProps & {
+    framework?: string;
+    path?: string | null;
+  }
+): null {
   useEffect(() => {
-    inject({ beforeSend, debug, mode });
-  }, [beforeSend, debug, mode]);
+    inject({
+      framework: props.framework || 'react',
+      ...(props.route !== undefined && { disableAutoTrack: true }),
```

Comment:
**question:** is your intention to disable auto tracking when the component has not `route` in its properties?

This is calling inject everytime a props has been modified. I guess it's ok because `inject` bails when there's already a script in `<head>`